scripts/dts/gen_defines.py

```diff
@@ -110,6 +110,28 @@ def main():
         write_chosen(edt)
         write_global_compat_info(edt)
 
+    with open(args.device_header_out, "w", encoding="utf-8") as dev_header_file:
+        print("#ifndef DEVICE_EXTERN_GEN_H", file=dev_header_file)
+        print("#define DEVICE_EXTERN_GEN_H", file=dev_header_file)
+        print("", file=dev_header_file)
+        print("#ifndef _ASMLANGUAGE", file=dev_header_file)
+        print("", file=dev_header_file)
+        print("#ifdef __cplusplus", file=dev_header_file)
+        print('extern "C" {', file=dev_header_file)
+        print("#endif", file=dev_header_file)
+        print("", file=dev_header_file)
+
+        for node in sorted(edt.nodes, key=lambda node: node.dep_ordinal):
+            print(f"extern const struct device __device_DT_{node.z_path_id};", file=dev_header_file)
```

Comment:
I'd rather not have this hard-code the representation of the external symbol.  Can we move the include of the generated file to the end of `device.h` and have this just use `DT_DEVICE_DECLARE();`?

Reply:
I like that, will rework this to use DT_DEVICE_DECLARE().